hack/verify-golint.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly VERSION="v1.46.2"
+readonly KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+cd "${KUBE_ROOT}"
+
+# See configuration file in ${KUBE_ROOT}/.golangci.yml.
+mkdir -p cache
+
+docker run --rm -v $(pwd)/cache:/cache -v $(pwd):/app --security-opt="label=disable" -e GOLANGCI_LINT_CACHE=/cache -w /app "golangci/golangci-lint:$VERSION" golangci-lint run
+
```

Comment:
Nit: extra lines?

Reply:
@astoycos it looks like if we don't specify it there's a default behavoir
https://golangci-lint.run/usage/configuration/#cache

Want to stick with being explicit and cleaning it up, or let it take the default $HOME/.cache?
https://pkg.go.dev/os#UserCacheDir